### PR TITLE
[API] Archivization toggling and filtration

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | Part of #11250
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php [hunk 1]
```diff
@@ -227,6 +236,25 @@ public function iSwitchTheWayShippingMethodsAreSortedByName(): void
...
+     * @When I filter archival shipping methods
+     */
+    public function iFilterArchivalShippingMethods(): void
+    {
+        $this->client->addFilter('exists[archivedAt]', true);
+        $this->client->filter();
+    }
+
+    /**
+     * @When I am browsing archival shipping methods
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php]:
```suggestion
     * @Given I am browsing archival shipping methods
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml [hunk 2]
```diff
@@ -25,6 +25,13 @@
             <tag name="api_platform.filter" />
         </service>
 
+        <service id="sylius.api.filter_archival_shipping_methods" parent="api_platform.doctrine.orm.exists_filter">
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml]:
```suggestion
        <service id="sylius.api.filter_archived_shipping_methods" parent="api_platform.doctrine.orm.exists_filter">
```
?

File: src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php [hunk 3]
```diff
@@ -227,6 +236,25 @@ public function iSwitchTheWayShippingMethodsAreSortedByName(): void
...
+     */
+    public function iAmBrowsingArchivalShippingMethods(): void
+    {
+        $this->client->index();
+        $this->client->addFilter('exists[archivedAt]', true);
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php]: I'm wondering, maybe we should have another API client just for index? Or filtering? Doing so, we could make more specified methods and skip technical details in the context 💃

Comment (lchrusciel) [reply src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php]: good idea 🎉

**Pull request merged**